src/odemis/driver/smaract.py

```diff
@@ -1804,14 +1798,15 @@ def __init__(self):
         self._pivot = SA_MC_Vec3()
 
         # Specify ranges
-        # Mirroring actual ranges
-        self._range = {}
-        self._range['x'] = (-1.6e-2, 1.6e-2)
-        self._range['y'] = (-1.5e-2, 1.5e-2)
-        self._range['z'] = (-1.e-2, 0.002)
-        self._range['rx'] = (-28, 28)
-        self._range['ry'] = (0, 0)
-        self._range['rz'] = (-28, 28)
+        # Make sure these ranges mirroring the simulator ranges (ie on the yaml file)
+        self._range = {
+            'x': (-1.6e-2, 1.6e-2),
+            'y': (-1.5e-2, 1.5e-2),
+            'z': (-1.0e-2, 0.003),
+            'rx': (-28, 28),
+            'ry': (0, 0),
+            'rz': (-28, 28),
+        }
```

Comment:
Instead of increasing the range to adjust to the ENZEL, we could automatically adjust it to fit the range requested.
Take a "rng" (optional) argument. Then update the _range to increase it if needed. Something like:
```python
for a, r in rng.items():
    orig_rng = self._range[a]
    self._range[a] = (min(orig_rng[0], r[0]), max(orig_rng[1], r[1]))
```
The MC_5DOF class can pass the range when it's a simulator. It's a bit cheating, but that's what the simulator is all about anyway ;-)